File: decompiled/3.19.1/be/ehealth/businessconnector/ehbox/api/domain/Document.java
package be.ehealth.businessconnector.ehbox.api.domain;

import be.ehealth.businessconnector.ehbox.api.domain.exception.EhboxBusinessConnectorException;
import be.ehealth.businessconnector.ehbox.api.domain.exception.EhboxBusinessConnectorExceptionValues;
import be.ehealth.businessconnector.ehbox.api.utils.SigningValue;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.exception.UnsealConnectorException;
import be.ehealth.technicalconnector.utils.ConnectorIOUtils;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.Serializable;
import java.text.MessageFormat;
import org.apache.commons.lang.ArrayUtils;
import org.bouncycastle.util.Arrays;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Document implements Serializable {
   private static final long serialVersionUID = 1L;
   private static final Logger LOG = LoggerFactory.getLogger(Document.class);
   private String title;
   private byte[] content;
   private String filename;
   private String mimeType;
   private transient SigningValue signing;
   private UnsealConnectorException expection;

   public final void getDocument(String fullpath) throws EhboxBusinessConnectorException {
      File file = new File(fullpath);
      if (file.isDirectory()) {
         file = new File(fullpath + System.getProperty("file.separator") + this.filename);
      }

      FileOutputStream fos = null;

      try {
         fos = new FileOutputStream(file);
         fos.write(this.content);
         fos.flush();
      } catch (IOException var8) {
         LOG.debug("\t## " + MessageFormat.format(EhboxBusinessConnectorExceptionValues.ERROR_EHBOX_DOCUMENT_OUTPUTSTREAM.getMessage(), file.toURI()));
         throw new EhboxBusinessConnectorException(EhboxBusinessConnectorExceptionValues.ERROR_EHBOX_DOCUMENT_OUTPUTSTREAM, var8, new Object[]{file.toURI()});
      } finally {
         ConnectorIOUtils.closeQuietly((Object)fos);
      }

   }

   public final byte[] getContent() throws UnsealConnectorException {
      if (this.content == null && this.expection != null) {
         throw this.expection;
      } else {
         return Arrays.clone(this.content);
      }
   }

   public final void setContent(byte[] content) {
      this.content = Arrays.clone(content);
   }

   /** @deprecated */
   @Deprecated
   public final void setContent(Byte[] content) {
      this.content = ArrayUtils.toPrimitive(content);
   }

   public final void setContent(InputStream inputStream) throws TechnicalConnectorException {
      this.content = ConnectorIOUtils.getBytes(inputStream);
   }

   public final String getFilename() {
      return this.filename;
   }

   public final void setFilename(String filename) {
      this.filename = filename;
   }

   public final SigningValue getSigning() {
      return this.signing;
   }

   public final void setSigning(SigningValue signing) {
      this.signing = signing;
   }

   public final String getMimeType() {
      return this.mimeType;
   }

   public final void setMimeType(String mimeType) {
      this.mimeType = mimeType;
   }

   public final String getTitle() {
      return this.title;
   }

   public final void setTitle(String title) {
      this.title = title;
   }

   public void setException(UnsealConnectorException expection) {
      this.expection = expection;
   }
}

File: src/main/kotlin/org/taktik/connector/business/ehbox/api/domain/Document.kt
/*
 *
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of FreeHealthConnector.
 *
 * FreeHealthConnector is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation.
 *
 * FreeHealthConnector is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with FreeHealthConnector.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.taktik.connector.business.ehbox.api.domain

import org.taktik.connector.business.ehbox.api.domain.exception.EhboxBusinessConnectorException
import org.taktik.connector.business.ehbox.api.domain.exception.EhboxBusinessConnectorExceptionValues
import org.taktik.connector.business.ehbox.api.utils.SigningValue
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.exception.UnsealConnectorException
import org.taktik.connector.technical.utils.ConnectorIOUtils
import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.io.InputStream
import java.io.Serializable
import java.text.MessageFormat
import org.apache.commons.lang.ArrayUtils
import org.bouncycastle.util.Arrays
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Document : Serializable {
    var title: String? = null
    private var content: ByteArray? = null
    var filename: String? = null
    var mimeType: String? = null
    @Transient
    var signing: SigningValue? = null
    private var expection: UnsealConnectorException? = null

    @Throws(EhboxBusinessConnectorException::class)
    fun getDocument(fullpath: String) {
        var file = File(fullpath)
        if (file.isDirectory) {
            file = File(fullpath + System.getProperty("file.separator") + this.filename)
        }

        var fos: FileOutputStream? = null

        try {
            fos = FileOutputStream(file)
            fos.write(this.content!!)
            fos.flush()
        } catch (var8: IOException) {
            LOG.debug("\t## " + MessageFormat.format(EhboxBusinessConnectorExceptionValues.ERROR_EHBOX_DOCUMENT_OUTPUTSTREAM.message, file.toURI()))
            throw EhboxBusinessConnectorException(EhboxBusinessConnectorExceptionValues.ERROR_EHBOX_DOCUMENT_OUTPUTSTREAM, var8, *arrayOf<Any>(file.toURI()))
        } finally {
            ConnectorIOUtils.closeQuietly(fos as Any?)
        }

    }

    @Throws(UnsealConnectorException::class)
    fun getContent(): ByteArray {
        return if (this.content == null && this.expection != null) {
            throw this.expection!!
        } else {
            Arrays.clone(this.content)
        }
    }

    fun setContent(content: ByteArray?) {
        this.content = content?.let { Arrays.clone(it) }
    }


    @Throws(TechnicalConnectorException::class)
    fun setContent(inputStream: InputStream) {
        this.content = ConnectorIOUtils.getBytes(inputStream)
    }

    fun setException(expection: UnsealConnectorException) {
        this.expection = expection
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(Document::class.java)
    }
}
